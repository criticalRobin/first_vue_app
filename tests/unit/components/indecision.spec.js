import Indecision from "@/components/Indecision.vue";
import { shallowMount } from "@vue/test-utils";

describe("Pruebas del componente Indecision", () => {
  let wrapper;
  let clgSpy;

  beforeEach(() => {
    wrapper = shallowMount(Indecision);
    clgSpy = jest.spyOn(console, "log");

    global.fetch = jest.fn(() =>
      Promise.resolve({
        json: () =>
          Promise.resolve({
            answer: "yes",
            forced: false,
            image:
              "https://yesno.wtf/assets/yes/4-c53643ecec77153eefb461e053fb4947.gif",
          }),
      })
    );

    jest.clearAllMocks();
  });

  test("El componente debe ser igual al snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Escribir en el input no debe disparar nada (console.log)", async () => {
    //vm es el objeto que contiene todos los metodos y propiedades del componente
    const getAnswerSpy = jest.spyOn(wrapper.vm, "getAnswer");
    const input = wrapper.find("input");
    await input.setValue("Hobis Mundo!!!");

    expect(clgSpy).toHaveBeenCalledTimes(1);
    expect(getAnswerSpy).not.toHaveBeenCalled();
  });

  test("Escribir el simbolo de '?' debe disparar el getAnswer", async () => {
    const getAnswerSpy = jest.spyOn(wrapper.vm, "getAnswer");
    const input = wrapper.find("input");
    await input.setValue("Tas aqui?");

    expect(getAnswerSpy).toHaveBeenCalled();
  });

  test("Pruebas en el getAnswer", async () => {
    await wrapper.vm.getAnswer();

    const image = wrapper.find("img");

    expect(image.exists()).toBeTruthy();
    expect(wrapper.vm.answer).toBe("SÃ­");
  });

  test("Pruebas en el getAnswer - Fallo del API", async () => {
    fetch.mockImplementationOnce(() => Promise.reject());

    await wrapper.vm.getAnswer();

    const image = wrapper.find("img");

    expect(image.exists()).toBeFalsy();
    expect(wrapper.vm.answer).toBe("Error al obtener la respuesta");
  });
});
