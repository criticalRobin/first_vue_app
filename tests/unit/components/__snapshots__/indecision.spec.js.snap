// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente Indecision El componente debe ser igual al snapshot 1`] = `
<!--v-if-->
<div class="bg-dark"></div>
<div class="indecision-container"><input type="text" placeholder="Hazme una pregunta">
  <p>No olvides finalizar con un (?)</p>
  <!--v-if-->
</div>
`;
